src/gui/attributeformconfig/qgsattributeformcontaineredit.cpp

```diff
@@ -78,7 +75,7 @@ void QgsAttributeFormContainerEdit::updateItemData()
 
   QgsOptionalExpression visibilityExpression;
   visibilityExpression.setData( QgsExpression( mVisibilityExpressionWidget->expression() ) );
-  visibilityExpression.setEnabled( mControlVisibilityGroupBox->isChecked() );
+  visibilityExpression.setEnabled( true );
```

Comment:
Any idea if there is a way to skip this? it  is not meant to ever become false so...

Reply:
It probably should be

    visibilityExpression.setEnabled( !mVisibilityExpressionWidget->currentText().isEmpty() );

There's no real way to avoid the enabled call without breaking API here.